content/blog/kubernetes-chart-v4/index.md

```diff
@@ -0,0 +1,1033 @@
+
+For resources in the “core” group, the empty string is used instead (for example: `/namespaces/test/Pod/pod-a`).
+
+### New-Style Pulumi Transformations
+
+Pulumi has a new way to transform component resources and their children, the `transforms`  options. The older
+`transformations` option doesn't work with multi-language components like Chart v4. See
+[Resource Option: transforms](content/docs/concepts/options/transforms.md) for more details.
+
+Note: you cannot change an object's namespace or name using a Pulumi transformation, and you cannot add or discard
+an object.
+
+Here's an example of using the `transforms` option to add the `pulumi.com/patchForce` annotation
+to a chart's resources.
+
+{{< chooser language "typescript,python" >}}
+
+{{% choosable language "javascript,typescript" %}}
+
+```typescript
+const applyPatchForceAnnotation = async (args: pulumi.ResourceTransformArgs) => {
+    switch(args.type) {
+        case "kubernetes:helm.sh/v4:Chart":
+            break;
+        default:
+            args.props.metadata.annotations = {
+                "pulumi.com/patchForce": "true",
+                ...args.props.metadata.annotations
+            }
+    }
+    return {
+        props: args.props,
+        opts: args.opts,
+    };
+};
+
+const ingressController = new kubernetes.helm.v4.Chart("ingresscontroller", {
+    chart: "nginx-ingress",
+    namespace: ingressNs.metadata.name,
+    repositoryOpts: {
+        repo: "https://helm.nginx.com/stable",
+    },
+    version: "0.14.1",
+}, {transforms: [applyPatchForceAnnotation]});
+```
+
+{{% /choosable %}}
+
+{{% choosable language python %}}
+
+```python
+def apply_patchforce_annotation(args: ResourceTransformArgs):
+    if not args.type_ == "kubernetes:helm.sh/v4:Chart":
+        if not 'metadata' in args.props:
+            args.props['metadata'] = {}
+        if not 'annotations' in args.props['metadata']:
+            args.props['metadata']['annotations'] = {}
+        args.props['metadata']['annotations']['pulumi.com/patchForce'] = 'true'
+
+    return ResourceTransformResult(
+        props=args.props,
+        opts=args.opts)
+
+ingresscontroller = kubernetes.helm.v4.Chart(
+    "ingresscontroller",
+    chart="nginx-ingress",
+    namespace=ingress_ns.metadata.name,
+    repository_opts=kubernetes.helm.v3.RepositoryOptsArgs(
+        repo="https://helm.nginx.com/stable",
+    ),
+    version="0.14.1",
+    opts=pulumi.ResourceOptions(transforms=[apply_patchforce_annotation])
+)
+```
+
+{{% /choosable %}}
+
+{{< /chooser >}}
+
+### Not Supported: Kubernetes Transformations
+
+Chart v4 does not support the `transformations` argument as seen in Chart v3, that facilitates a
+Kubernetes-centric transformation and/or discarding of objects from the rendered manifest.
+
+One alternative is to use use Pulumi transformations to transform the object and resource options.
+Another is to use post-rendering, which we'll cover next.
+
```

Comment:
I think it might make sense to move the notes on limitations/not supported (this and hooks it looks like) to a separate section at the end to distinguish them from the new features